src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -68,12 +68,17 @@ public synchronized void reportResults(List<Survey> newSurveys, List<InAppNotifi
             }
         }
 
+        if (variants.length() > 0) {
+            newContent = true;
+        }
+
         mVariants = variants;
```

Comment:
can you please move this line into the if statement above? this is because we have a function checks if `mVariants` is null. `variants` will be an empty string (not null) if there's no variant, which breaks this check.
```
    public synchronized boolean hasUpdatesAvailable() {
        return (! mUnseenNotifications.isEmpty()) || (! mUnseenSurveys.isEmpty()) || mVariants != null;
    }
```